src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -0,0 +1,466 @@
+use crate::commands::beta::generate::context::{CodeGenContext, DataModel};
+use crate::commands::beta::generate::result::{invalid_data_model_error, CodeGenResult};
+use crate::commands::beta::generate::utils::{Field, Import, Language};
+use convert_case::{Case, Casing};
+use ion_schema::isl::isl_constraint::{IslConstraint, IslConstraintValue};
+use ion_schema::isl::isl_type::IslType;
+use ion_schema::isl::isl_type_reference::IslTypeRef;
+use ion_schema::isl::IslSchema;
+use std::collections::HashMap;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+use tera::Context;
+
+pub(crate) struct CodeGenerator;
+
+impl CodeGenerator {
+    /// Maps the given type name to a native type based on programming language
+    pub fn map_to_base_type(name: &str, language: &Language) -> String {
+        match (name, language) {
+            ("int", Language::Java) => "int".to_string(),
+            ("string" | "symbol" | "text", _) => "String".to_string(),
+            ("int", Language::Rust) => "i64".to_string(),
+            ("float", Language::Rust) => "f64".to_string(),
+            ("float", Language::Java) => "float".to_string(),
+            ("bool", Language::Rust) => "bool".to_string(),
+            ("bool", Language::Java) => "boolean".to_string(),
+            ("blob" | "clob" | "lob", Language::Rust) => "Vec<u8>".to_string(),
+            ("blob" | "clob" | "lob", Language::Java) => "byte[]".to_string(),
+            ("decimal" | "timestamp" | "number", _) => {
+                unimplemented!("Decimal, Number and Timestamp aren't support yet!")
+            }
+            ("list" | "struct" | "sexp" | "document", _) => {
+                unimplemented!("Generic containers aren't supported yet!")
+            }
+            (_, _) => name.to_case(Case::UpperCamel),
```

Comment:
I think we need an `IslType` enum that looks something like this:
```rust
enum IslType {
  Null,
  Int,
  Bool,
  // ...
  Lob,
  Number,
  Document,
  Nothing,
  SchemaDefined(String), // name
}
```
This allows `Language` variants to benefit from `match`'s exhaustiveness check and explicitly calls out the `SchemaDefined` type case.